app/dot/cpp/repository/services/RepositoryService.java

```diff
@@ -17,59 +19,82 @@
   private final Config config;
 
   @Inject
-  public RepositoryController(Config config) {
+  public RepositoryService(Config config) {
     this.config = config;
   }
 
-  public Result init(List<Class<? extends BaseEntity>> entities) {
-    createDatabase(entities);
-    return ok("init complete");
-  }
-
-  public void createDatabase(List<Class<? extends BaseEntity>> entities) {
+  public void createCollections(List<Class<? extends BaseEntity>> entities, boolean withHistory) {
     try (final MongoClient mongoClient = new MongoClient()) {
-      var database = mongoClient.getDatabase(config.getString("db.name"));
-      createCollections(entities, database);
+      var database = mongoClient.getDatabase(config.getString("morphia.database"));
+      createCollections(database, entities, withHistory);
     }
   }
 
   private void createCollections(
-      List<Class<? extends BaseEntity>> entities, MongoDatabase database) {
+      MongoDatabase database, List<Class<? extends BaseEntity>> entities, boolean withHistory) {
     final var schemaGeneratorConfig = getSchemaGeneratorConfig();
     entities.forEach(
         entity -> {
           var schema = getSchema(entity, schemaGeneratorConfig);
           var validationOptions =
               new ValidationOptions().validator(Filters.jsonSchema(Document.parse(schema)));
 
-          if (isCollectionInDatabase(entity.getSimpleName(), database)) {
-            logger.debug("already exists");
-
-            database.runCommand(
-                new Document("collMod", entity.getSimpleName())
-                    .append("validator", Filters.jsonSchema(Document.parse(schema)))
-                    .append("validationLevel", "strict"));
-            // default values are added by setting the variable in its respective class with an
-            // initial value
-          } else {
-            database.createCollection(
-                entity.getSimpleName(),
-                new CreateCollectionOptions().validationOptions(validationOptions));
+          createCollection(database, entity.getSimpleName(), schema, validationOptions);
+
+          if (withHistory) {
+            createCollection(
+                database, entity.getSimpleName() + "_history", schema, validationOptions);
+
+            // todo uncomment after history is implemented throughout the application
+            // createIndex(database, entity.getSimpleName(), "trackingId", true);
+            createIndex(database, entity.getSimpleName() + "_history", "trackingId", false);
           }
 
           logger.debug("{}", schema);
         });
   }
 
+  private void createIndex(
+      MongoDatabase database, String entityName, String fieldName, Boolean unique) {
+    final var collection = database.getCollection(entityName);
+
+    for (final var index : collection.listIndexes()) {
+      if (index.getString("name").equals(fieldName + "_idx")) {
+        collection.dropIndex(fieldName + "_idx");
+        break;
+      }
+    }
```

Comment:
Why are we doing this? Won't the creation of the index happen only once (at table creation)?

Reply:
The old index might be conflicting with the new one, so it can't be simply rewritten. Dropping the old one first fixes the issue. This method will be useful in the future as well when we create indexes for fields that get looked up frequently.